shared/Constants/Constants.F90

```diff
@@ -0,0 +1,21 @@
+module MAPL_Constants
+
+use, intrinsic :: iso_fortran_env, only: REAL64, REAL32
+use MAPL_InternalConstantsMod, disabled => MAPL_UNDEF !For temporary backward compatibility
```

Comment:
@tclune is `disabled` a new Fortran thing? Not sure I've seen it before...

Reply:
Ahh. @Gvilla1000-nasa explained this...

Reply:
it is not. This is just a proxy rename to avoid conflicts between MAPL_UNDEF from BaseMod and MAPL_UNDEF from MAPL_Constant. I had to do it this way because several grid components external to MAPL still use MAPL_UNDEF from BaseMod and not doing this caused conflicts.

Reply:
I don't think this solution helps us much.  If we can't get MAPL_UNDEF from MAPL_Constants, it defeats the purpose of putting it there.   Not saying I have a better solution though.

Reply:
Based on the meetings we had, couldn't this change be done when MAPL moves to the next major revision at which point users will have to change their code to use MAPL_Constants rather than BaseMod? This was my motivation for doing it this way. Paving the road for that point in time.

Reply:
I'd rather just not export it in Constants at all than under a different name.  

@mathomp4 are the external uses of BaseMod all in DAS?  I thought we did a sweep a while back to make those all MAPL, but maybe this was just the GCM or just the targets in cmake?

Not really liking any of our options here.

Reply:
I did a quick search for uses of `MAPL_UNDEF` from BaseMod and found the following results:
```
Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GMIchem_GridComp/GMI_GridComp/GmiChemistry/photolysis/CloudJ/CLD_SUB_MOD.F90
24:      use MAPL_BaseMod, only: MAPL_UNDEF

Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GMIchem_GridComp/GMI_GridComp/GmiChemistry/photolysis/CloudJ/FJX_SUB_MOD.F90
29:      use MAPL_BaseMod,      only: MAPL_UNDEF

Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/@GEOSchem_GridComp/GMIchem_GridComp/GMI_GridComp/GmiChemistry/photolysis/fast_JX65/fastJX65_mod.F90
7:      use MAPL_BaseMod, only: MAPL_UNDEF

Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSmoist_GridComp/cloudnew.F90
23:   use MAPL_BaseMod,      only: MAPL_UNDEF

Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSmoist_GridComp/cldmacro.F90
23:   use MAPL_BaseMod,      only: MAPL_UNDEF

Shared/@GMAO_Shared/GEOS_Util/post/ec_eta2fv.F
408:      use MAPL_BaseMod, only: MAPL_UNDEF

Shared/@GMAO_Shared/GEOS_Util/post/ec_prs2eta.F
1761:      use MAPL_BaseMod, only: MAPL_UNDEF

Shared/@GMAO_Shared/GEOS_Util/post/gg2fv.F
544:      use MAPL_BaseMod, only: MAPL_UNDEF
990:      use MAPL_BaseMod, only: MAPL_UNDEF

Shared/@GMAO_Shared/GEOS_Util/post/gg2eta.F
236:      use MAPL_BaseMod, only: MAPL_UNDEF

Shared/@GMAO_Shared/GEOS_Util/post/mpi_util.F
716:      use MAPL_BaseMod, only: MAPL_UNDEF
841:      use MAPL_BaseMod, only: MAPL_UNDEF

Shared/@GMAO_Shared/GEOS_Util/post/era5_prs2eta.F
1829:      use MAPL_BaseMod, only: MAPL_UNDEF
```

However, curiously enough, when I removed the `MAPL_UNDEF` declaration from Base.F90 and made it so that the MAPL code used `MAPL_UNDEF` from `MAPL_Constants` rather than `MAPL_BaseMod`, the compilation didn't fail at any of them. It did fail for  `src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSsuperdyn_GridComp/@FVdycoreCubed_GridComp/AppGridCreate.F90` because it couldn't find `pi` from `MAPL_ConstantsMod` which I find odd because why did changing MAPL_UNDEF impact pi?

Reply:
I'm almost tempted to update GMAO_shared to use MAPL_Constants now, but there is a risk that the GCM will need an update there for other reasons and then need the newer MAPL and ...

It might be worth double checking things with the FV "pi mystery".     Maybe some module did not correctly recompile for reasons that I cannot see.    Or maybe Base is also exporting Pi for reasons that are not immediately obvious?

Make clean would be the safest thing to try.   Be sure to read any error messages carefully they might not quite indicate what you think ...

Reply:
My apologies for the delay. So this was a doozy to fix but the fix turned out to be very simple but perplexing to me. The FV "pi mystery" was fixed by changing `use MAPL_ConstantsMod` to `use MAPL_ConstantsMod, only: MAPL_PI_R8` inside `base/MAPL_Mod.F90`. Not sure why this fix was needed only when `MAPL_UNDEF` was removed from `MAPL_BaseMod` though. After making this fix, no other compilation errors were reported.  I re-compiled GEOSgcm with intel and gnu in both debug and release (making clean everything each time) and no issues were seen. Running a quick intel debug run still showed zero diff to previous runs. Should I commit the removal of `MAPL_UNDEF` from `BaseMod`? I'm concerned about what sort of effects it may cause at runtime or compilation of other components that were not tested since there are still a lot of `MAPL_UNDEF` from `BaseMod` uses in GridComp and GMAO_Shared.

Reply:
My only strict requirement for the current PR is that it is provably 0 diff for GCM and, as much as we can test, the DAS.  (The latter is probably just source inspection for now unfortunately.)

But ... I want to go ahead and start a MAPL3 branch (name TBD) where we will start putting our desired changes that break backward compatibility.   (@mathomp4 - you might want to consider this in terms of the CODATA constants.)

Reply:
Ok. Then as things stand now, it is 0 diff (without removing `MAPL_UNDEF` from `BaseMod`). The question I have is should I revert the addition of `MAPL_UNDEF` to `MAPL_Constants` since I pretty much hacked it to be disabled through the use of the `disabledvar => MAPL_UNDEF` rename? The change proposed in the last comment can be left to the new branch since a much better fix would be for the FV to use the `mapl.constants` library.

Reply:
Yes - revert.  Save that for the 3.0 branch.